var/spack/repos/builtin/packages/adept/package.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Adept(AutotoolsPackage):
+    """Combined array and automatic differentiation library in C++."""
+
+    homepage = "https://www.met.reading.ac.uk/clouds/adept/"
+    url = "https://www.met.reading.ac.uk/clouds/adept/adept-2.1.1.tar.gz"
+
+    maintainers("jehicken")
+
+    license("Apache-2.0", checked_by="jehicken")
+
+    version("2.1.1", sha256="0cef334e82df4526d3761bdd8319a63e7582c96b2f1cc88391729018b4825c47")
+
+    variant("debug", default=False, description="Enable debugging information")
+    variant("blas", default=False, description="Enable Adept's native arrays using Openblas")
+    variant("lapack", default=False, description="Enable Adept's native arrays using Lapack")
+
+    depends_on("cxx", type="build")
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+    depends_on("m4", type="build")
+
+    depends_on("openblas", when="+blas")
+    depends_on("netlib-lapack", when="+lapack")
+
+    def autoreconf(self, spec, prefix):
+        autoreconf("--install", "--verbose", "--force")
+
+    def configure_args(self):
+        args = []
+
+        if self.spec.satisfies("+debug"):
```

Comment:
Setting CXXFLAGS is better done through a `flag_handler` and should allow passthrough of user-specified flags (`spack install adept cxxflags=-fno-unroll-loops` for example).

Reply:
Thanks @wdconinc.  Can you point me to an example of how to use the `flag_handler`?  I found the documentation [here](https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags), but an example that is closer to my use case would help.  Or, is Spack's default flag handling sufficient for this case, and I can just delete the conditional related to `"+debug"`?

Reply:
To be honest, in this case where it is simply a matter of injecting -g and nothing else I would let users inject this with cxxflags as part of the spec. It's more useful to do this with a variant when you also set other flags or need to define some preprocessor variables.

Reply:
Ok, in that case, I have removed the `debug` variant and this conditional.  Thanks again!